spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerIntegration2Tests.java

```diff
@@ -318,6 +319,86 @@ public Channel answer(InvocationOnMock invocation) throws Throwable {
 
 		assertTrue(latch.await(10, TimeUnit.SECONDS));
 		assertTrue(networkGlitch.get());
+
+		container.stop();
+	}
+
+	@Test
+	public void testRestartConsumerOnConnectionLossDuringQueueDeclare() throws Exception {
+		this.template.convertAndSend(queue.getName(), "foo");
+
+		ConnectionFactory connectionFactory = new CachingConnectionFactory("localhost", BrokerTestUtils.getPort());
```

Comment:
When I was writing that test for the `basicQos` I worried about the new `connection` for more than one `connectionFactory.createConnection()`. But now I see that it doesn't matter, because default mode is `CacheMode.CHANNEL`, so we share the single connection.
Hence your solution is OK. 
LGTM, merging with simple polishing for those `@since`, `@Copyright` and rebase to `master`